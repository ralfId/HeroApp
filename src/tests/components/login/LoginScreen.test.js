import React from 'react';
import { mount } from 'enzyme'
import { LoginScreen } from "../../../components/login/LoginScreen";
import { MemoryRouter } from 'react-router-dom';
import { AuthContext } from '../../../auth/AuthContext';
import { types } from '../../../types/types';

describe('pruebas en LoginScreen.js', () => {

    const valueContext = {
        dispatch: jest.fn(),
        user: { logged: false }
    }

    const historyMock = {
        replace: jest.fn()
    }
    const wrapper = mount(
        <AuthContext.Provider value={ valueContext}>
            <MemoryRouter>
                <LoginScreen history={historyMock}/>
            </MemoryRouter>
        </AuthContext.Provider>
    )
    test('debe  mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot()
    });

    test('debe de realizar el disparch  y la navegacion', () => {
        
        const handleClick = wrapper.find('button').prop('onClick');
        
        handleClick();

        expect(valueContext.dispatch).toHaveBeenCalledWith({type: types.login, payload: {name: 'Rafael'}})
        expect(historyMock.replace).toHaveBeenCalledWith('/')

        //simulando el localStorage
        localStorage.setItem('lastPath', '/dc');
        handleClick();

        expect(historyMock.replace).toHaveBeenCalledWith('/dc')




        
    });
});