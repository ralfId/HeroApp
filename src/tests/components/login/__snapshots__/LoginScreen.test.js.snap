// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en LoginScreen.js debe  mostrarse correctamente 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login Screen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    LOGIN
  </button>
</div>
`;
